src/registry/artifactory.rs

```diff
@@ -24,24 +23,50 @@ use semver::Version;
 use serde::Deserialize;
 use url::Url;
 
+/// The policy for validating artifactory server certificates
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Default)]
+pub enum CertValidationPolicy {
+    /// Validate the certificate
+    #[default]
+    Validate,
+
+    /// Do not validate the certificate
+    NoValidation,
+}
+
 /// The registry implementation for artifactory
 #[derive(Debug, Clone)]
 pub struct Artifactory {
-    registry: RegistryUri,
+    registry: url::Url,
     token: Option<String>,
     client: reqwest::Client,
 }
 
 impl Artifactory {
     /// Creates a new instance of an Artifactory registry client
-    pub fn new(registry: RegistryUri, credentials: &Credentials) -> miette::Result<Self> {
+    ///
+    /// # Arguments
+    /// * `registry` - The registry URI
+    /// * `credentials` - The credentials to use for the registry
+    /// * `cert_validation_policy` - The policy for validating artifactory server certificates
+    pub fn new(
+        registry: url::Url,
+        credentials: &Credentials,
+        cert_validation_policy: CertValidationPolicy,
```

Comment:
```suggestion
        policy: CertValidationPolicy,
```

Nit! Personal opinion: the type name conveys enough info here

Reply:
a6a41b4a48640adb766999e3c145db5895e84059